### ISSUE TITLE
borg mount throws "Transport endpoint is not connected"... yeat another one

### ISSUE THREAD

**Issue by boelle** [borgbackup/borg#7672]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

YES

## Is this a BUG / ISSUE report or a QUESTION?

ISSUE, but sure i goof up somehow

#### Your borg version (borg -V).

borg 1.2.3

#### Operating system (distribution) and version.

Openmediavault 6.4.5-1 (Shaitan)

#### Hardware / network configuration, and filesystems used.

HP Proliant N40l Microserver
EXT4 

#### How much data is handled by borg?

Not sure, but ~500 GB

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

borg mount -o versions ssh://root@100.121.165.103/srv/mergerfs/Data/homedir/Malcolm/Borgbackup/Backup-Bo-OMV_NOVA /srv/mergerfs/Data/Pool1-Backup

+

rsync -auvc --progress /srv/mergerfs/Data/Pool1-Backup/srv/mergerfs/Pool1/Public-Test/Hannah /srv/mergerfs/Data/Backup/srv/mergerfs/Pool1/Public-Test

## Describe the problem you're observing.

sooner or later i get "Transport endpoint is not connected" when i use Rsync to restore the backup

i know that there is borg extract but there is no explanation on how to do a "merged" extract like i can do when mounting as "versions" and then rsync to a temp folder

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

just wait a bit and it will happen... sometimes it happens within sec's and sometime minutes, rarely an hour will pass

#### Include any warning/errors/backtraces from the system logs

let me know if i miss anything and i will add to FIRST POST

**Comment by boelle** [comment]
seems so far that either 1 job at the time... or the speed has something to do with it

**Comment by ThomasWaldmann** [comment]
That's not a `borg mount` problem, but your repo files are corrupted.

First check if hardware / OS works correctly, then you can try `borg check --repair`.

**Comment by boelle** [comment]
yeah.. i skipped a directory and the problems is gone

i will try and narrow it down some more

can i skip the --repair so it only checks and say what is wrong where?

**Comment by ThomasWaldmann** [comment]
Yes, running without `--repair` will mostly only show what it finds. But as I said, first fix hw/OS or it might get worse.

**Comment by boelle** [comment]
i'm a great noob, so can you tell me how i check that?

**Comment by ThomasWaldmann** [comment]
Maybe in this order:

You can use `memtest86+` on a USB stick and check your RAM. Run at least 1 full pass.

`smartctl -a /dev/sdX` and `smartctl -t long /dev/sdX` (and after that has finished, again with `-a`) could be used to check the HDD/SSD.

Linux `fsck` can be used to check the filesystem.

If all is fine, try `borg check --repair`.

**Comment by boelle** [comment]
Oki. I run these test automated once a month

Except memtest and fsck...


lør. 24. jun. 2023 12.46 skrev TW ***@***.***>:

**Comment by boelle** [comment]
Oki. I run these test automated once a month

Except memtest and fsck...

**Comment by ThomasWaldmann** [comment]
RAM errors can have pretty bad consequences. OTOH, the N40L supports ECC memory (and by default also comes with ECC memory), so the memory system should have a decent quality. OTOH, the N40L is pretty old, so it could be also an age related malfunction of some component.

**Comment by boelle** [comment]
is it possible to extract files from a mounted repro but skipping corrupted ones?

now i'm looking for a simple way to extract as much i can from there that is know good and just skip the corrupt files

right now i'm doing a check --verify-data > check.txt to have a file with all the errors
my plan is to clean up that file so i just have a list of files that are not "clean"



but are there a faster and better way?

**Comment by ThomasWaldmann** [comment]
Guess `borg mount` (like most borg command) expects a consistent/undamaged borg repo.

Hopefully your hardware works ok now, otherwise that borg check run might not work well (and `--repair` could cause more damage). In case of doubt, I guess I rather would have moved the repo to a known-good machine before working on it.

To redirect the log output, you need to redirect stderr.

After `borg check --repair` fixed all issues, you should be able to use `borg mount`. IIRC, if a file is damaged (has replacement chunks) it does not let you read the file, except if you give a special mount option.

**Comment by someone-somenet-org** [comment]
Could it be that you temp. lose internet connectivity during your rsync operation?

FUSE throws ``Transport endpoint is not connected``-errors at me when i work with for example sshfs and I lose my wifi connectivity and after re-establishing ssh fails to resume my sessions. As in: its not a borg backup issue for me.

``borg mount`` uses FUSE afaik.

**Comment by noppel** [comment]
I have the same problem ("endpoint not connected") requiring me to remount the file system. I have not been able to generate debug output. The problem does not occur on a specific directory/file. It happens randomly. I do not know how to continue from here.

**Comment by ThomasWaldmann** [comment]
@noppel I you could make it a habit to always use `borg mount -f ...` in another terminal, you could maybe get a traceback some time and the issue could be found.

**Comment by boelle** [comment]
and maybe do a borg check to be sure backup is not corrupted

**Comment by noppel** [comment]
> @noppel If you could make it a habit to always use `borg mount -f ...` in another terminal, you could maybe get a traceback some time and the issue could be found.

I have added --foreground (-f was wrong it seemed) and --debug. This is the result I got before the borg command returned:

fuse: _process_archive completed in 114.9 s for archive 2023-05-10_21:00
Calling fuse_session_destroy
RepositoryCache: current items 36751, size 2.11 GB / 2.15 GB, 501699 hits, 53425 misses, 8838 slow misses (+211.5s), 25512 evictions, 0 ENOSPC hit
RemoteRepository: 3.58 MB bytes sent, 3.96 GB bytes received, 67554 messages sent

**Comment by noppel** [comment]
> The output you got seems normal, I don't see anything unexpected there.

The process is exiting, tihs should not happen. It should continue running.

**Comment by boelle** [comment]
did you do a borg check to be sure backup is not corrupted

**Comment by ThomasWaldmann** [comment]
The problems usually can be debugged by running `borg mount --foreground ...`, so a traceback gets visible in case borg's FUSE code crashes.

If there is no traceback and `borg mount --foreground ...` just terminates, the fs was either unmounted (then termination is expected behaviour) or there is a yet unknown bug in borg's FUSE code (for this, please open a new issue).

**Comment by boelle** [comment]
I gave up 2 years ago and used rsync instead. On a tailscale connection, worked otb
